pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1842,6 +1846,12 @@ public void createPersistentTopic0(final String topic, boolean createIfMissing,
                         @Override
                         public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                             try {
+                                if (topicFuture.isCompletedExceptionally()) {
+                                    // Don't close the managed ledger because next time the topic is accessed, the
+                                    // managed ledger will be created again. The managed ledger will be removed if the
+                                    // ownership has been changed.
+                                    return;
+                                }
                                 PersistentTopic persistentTopic = isSystemTopic(topic)
                                         ? new SystemTopic(topic, ledger, BrokerService.this)
                                         : newTopic(topic, ledger, BrokerService.this, PersistentTopic.class);
```

Comment:
Maybe we can `checkDeduplicationStatus` first before `checkReplication` to avoid creating replicator.